elastic/fleet-server
# Seperate unenrollment canceller from policy

## What is the problem this PR solves?

If the unenroll timeout is set, then it applies to instances even after the timeout is cleared (in the fleet-ui).

## How does this PR solve the problem?

Track the canceller functions for the unenroller method separately from the policy struct.
This ensures that any running unenroller for a policy is cancelled before a new one starts.
Additionally add an `unenroller_uuid` attribute to (related) loggers to help track the issue. 

## How to test this PR locally

1. enroll an agent with the default policy
2. set timeout to 120 (through ui)
3. stop agent and observe that it goes to inactive after the timeout value
4. clear timeout value
5. re-enroll agent, and see that it is healthy in the ui
6. stop agent

Without the fix after the final stop the agent is set to inactive after the old timeout.
With the fix it should appear as offline.

## Checklist

- [] ~~I have made corresponding changes to the documentation~~
- [] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works **tested manually**
- [] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Related issues

- Closes #735
